### PR TITLE
Added new riddle: Renaming dataframe columns

### PR DESCRIPTION
**Pull request by RichardPilbery**
My first PR in response to Issue #5 for Hacktoberfest!

### CODE REVIEW

File: inst/example/questions/q4.riddlr.Rmd [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+difficulty: 0.1
+tags: 
+  - dataframes
+details: >
+  Give a dataframe meaningul column names
+editor_options: 
+  chunk_output_type: console
+---
+
+You're provided with a data frame, `df`, containing important data. However, the column names are generic and provide no clue as to the variable. You have a vector of strings that provide meaningful column names, `helpful_column_names`. Replace the column names of data frame `df` with the vector `helpful_column_names`.
```
Comment (dgkf) [inline inst/example/questions/q4.riddlr.Rmd:13]: I like it! Good idea. This is a great and common activity, and one that can be surprisingly complicated. It would be great if we probed some of these more difficult situations. 

As one small tweak, perhaps we can rename columns by name instead of position - meaning that we can't just assign `names<-` a vector directly, as the correct order will be unknown. 

Maybe we could provide the desired renamings as a matrix, mapping old names to new names. This way, to solve the problem, someone would have to get the current names, do some computation on them to remap them appropriately and set the new column names. In my experience, this is a very common activity, and one that really benefits from programmatic solutions instead of assuming columns will always be in a particular order. 

```r
df <- data.frame(a = 1:3, b = 1:3, c = 1:3)
column_name_map <- cbind(old = c("c", "b", "a"), new = c("last", "middle", "first"))
```

As well, let's try to find some edge cases to test in the latter tests. The purpose of these is to make sure the implementation has addressed behaviors that might not be obvious from the question, but these currently aren't testing any new behaviors. If we got the route of providing column names as a named vector that maps from current column names, then we run into some more difficult edge cases to handle. You might consider testing cases where:

1. Multiple columns have the same name
2. Some columns have no name (which gets converted to NA)
3. The data.frame has no column names, in which case all column names should be treated as `NA` for mapping them to new names
4. Columns have difficult characters like spaces ("example name"), periods ("example.name") or escape characters ("example\nname") in the name.

So a complicated scenario might look like this:

```r
# prep input data.frame
df <- data.frame(a = 1:3, b = 1:3, c = 1:3, d = 1:3, e = 1:3)
names(df) <- c("a", "b", "b", "c")  # 5th column will have NA name

old_names <- c(NA, "b", "c", "a")
new_names <- c("last", "middle", "middle", "first")
column_name_map <- cbind(old = old_names, new = new_names)
```

Let me know what you think.

**Review by dgkf — CHANGES_REQUESTED** [review]
Thanks for adding this! I think it's a great and common topic. I added a comment to tweak the overall promise in a way that allows us to add some more interesting edge cases that would introduce slightly more challenging edge cases.

File: inst/example/questions/q4.riddlr.Rmd (hunk 1 above)
Comment (RichardPilbery) [reply inst/example/questions/q4.riddlr.Rmd:13]: I think there is potential for including both scenarios, the straightforward and complex. I deal with some pretty messy data, but haven't needed to resort to the methods you've outlined. tidyverse functions definitely help with missing column names etc. and it is unusual for me not to review the incoming data and so know how it is formatted. 

There is a balance between making these riddles intellectually challenging vs. putting off learners.

**Review by dgkf — APPROVED** [review]
